lib/processor.js

```diff
@@ -107,18 +130,20 @@ exports.validate = (filePath) => {
 exports.download = (uploadUrl) => {
   let fileName = path.basename(url.parse(uploadUrl || 'unknown').pathname);
   let writeStream = fs.createWriteStream(`/tmp/${fileName}`)
+  let result = {name: fileName, localPath: `/tmp/${fileName}`};
 
   return Q.Promise((resolve, _reject) => {
     let reject = err => { err.fromDownload = true; _reject(err) };
     writeStream.on('error', err => reject(err));
-    writeStream.on('close', () => resolve({name: fileName, localPath: `/tmp/${fileName}`}));
+    writeStream.on('close', () => resolve(result));
 
     if (!uploadUrl) {
       getRemoteFile = reject(new Error('No url set for event'));
     } else if (uploadUrl.startsWith('s3://')) {
       let m = uploadUrl.match(/s3:\/\/([^\/]+)\/(.*)/);
-      let params = {Bucket: m[1], Key: m[2]};
-      s3.getObject(params)
+      result.s3Bucket = m[1];
+      result.s3Key = m[2];
+      s3.getObject({Bucket: m[1], Key: m[2]})
         .createReadStream()
         .on('error', err => reject(err))
         .pipe(writeStream);
```

Comment:
Oddly, there's no way to get the content-type (or any S3 metadata) while piping the response.  Would have to load the whole file into an in-memory buffer, or do a 2nd request to `s3.headObject`.

Since we can get the content-type from ffprobe anyways, i thought this was fine for now.